// @azure/ai-projects/dist/commonjs/connections/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.getConnectionsOperations = getConnectionsOperations;
const tslib_1 = require("tslib");
const connections_js_1 = require("./connections.js");
tslib_1.__exportStar(require("./inputOutput.js"), exports);
function getConnections(context) {
    return {
        getWorkspace: (options) => (0, connections_js_1.getWorkspace)(context, options),
        listConnections: (options) => (0, connections_js_1.listConnections)(context, options),
        getConnection: (connectionName, options) => (0, connections_js_1.getConnection)(context, connectionName, options),
        getConnectionWithSecrets: (connectionName, options) => (0, connections_js_1.getConnectionWithSecrets)(context, connectionName, options),
    };
}
/**
 * Get the connections operations
 * @returns The connections operations
 **/
function getConnectionsOperations(context) {
    return Object.assign({}, getConnections(context));
}
