// @azure/ai-projects/dist/browser/aiProjectsClient.js
import { getAgentsOperations } from "./agents/index.js";
import { getConnectionsOperations } from "./connections/index.js";
import createClient from "./generated/src/projectsClient.js";
import { getTelemetryOperations } from "./telemetry/index.js";
/**
 * The Azure AI Projects client
 */
export class AIProjectsClient {
    /*
     * @param endpointParam - The Azure AI Foundry project endpoint, in the form `https://<azure-region>.api.azureml.ms` or `https://<private-link-guid>.<azure-region>.api.azureml.ms`, where <azure-region> is the Azure region where the project is deployed (e.g. westus) and <private-link-guid> is the GUID of the Enterprise private link.
     * @param subscriptionId - The Azure subscription ID.
     * @param resourceGroupName - The name of the Azure Resource Group.
     * @param projectName - The Azure AI Foundry project name.
     * @param options - the parameter for all optional parameters
     */
    constructor(endpointParam, subscriptionId, resourceGroupName, projectName, credential, options = {}) {
        const connectionEndPoint = `https://management.azure.com/subscriptions/${subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/${projectName}`;
        this._client = createClient(endpointParam, subscriptionId, resourceGroupName, projectName, credential, options);
        this._connectionClient = createClient(endpointParam, subscriptionId, resourceGroupName, projectName, credential, Object.assign(Object.assign({}, options), { endpoint: connectionEndPoint }));
        this._telemetryClient = createClient(endpointParam, subscriptionId, resourceGroupName, projectName, credential, Object.assign(Object.assign({}, options), { apiVersion: "2020-02-02", endpoint: "https://management.azure.com" }));
        this.agents = getAgentsOperations(this._client);
        this.connections = getConnectionsOperations(this._connectionClient);
        this.telemetry = getTelemetryOperations(this._telemetryClient, this.connections);
    }
    /**
     * Creates a new instance of AzureAIProjectsClient
     * @param connectionString - Connection string with the endpoint, subscriptionId, resourceGroupName, and projectName
     * @param credential - The credential to use
     * @param options - The parameter for all optional parameters
     */
    static fromConnectionString(connectionString, credential, 
    // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
    options = {}) {
        const { endpointParam, subscriptionId, resourceGroupName, projectName } = AIProjectsClient.praseConnectionString(connectionString);
        return new AIProjectsClient(endpointParam, subscriptionId, resourceGroupName, projectName, credential, options);
    }
    static praseConnectionString(connectionString) {
        const parts = connectionString.split(";");
        return {
            endpointParam: `https://${parts[0]}`,
            subscriptionId: parts[1],
            resourceGroupName: parts[2],
            projectName: parts[3],
        };
    }
}
